databricks/koalas/base.py

```diff
@@ -676,4 +676,4 @@ def shift(self, periods=1, fill_value=None):
             shifted_col.isNull() | F.isnan(shifted_col), fill_value
         ).otherwise(shifted_col)
 
-        return self._with_new_scol(col).alias(self.name)
+        return self._with_new_scol(col).rename(self.name)
```

Comment:
Seems like this is not enough to make this working with Index. I am going to investigate and make another fix to make all of those work with Index as well.